temporal-sdk/src/main/java/io/temporal/internal/worker/ActivityPollTask.java

```diff
@@ -110,6 +110,7 @@ public ActivityTask poll() {
               ProtobufTimeUtils.toM3Duration(
                   response.getStartedTime(), response.getCurrentAttemptScheduledTime()));
       isSuccessful = true;
+      return new ActivityTask(response, pollSemaphore::release);
```

Comment:
Paired with 
```
if (!isSuccessful) pollSemaphore.release();
```
in finally, this location of `return` is cleaner for the reader.